### DISCUSSION TITLE
ComboBox not work when AddHandler(PointerPressedEvent, voidMethod, RoutingStrategies.Bubble) is enabled

### DISCUSSION THREAD

**Discussion by henrrymoraes83** [AvaloniaUI/Avalonia#12685] [Q&A] [answered]
When I add the method AddHandler(PointerPressedEvent, Button_OnPointerPressed, RoutingStrategies.Bubble); inside the constructor, the ComboBox inside the AXAML doesn't work. Could anyone explain the reason?

MainWindow.axaml.cs
```
using Avalonia.Controls;
using Avalonia.Input;
using Avalonia.Interactivity;
namespace ComboBox_Test.Views;

public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
        mainWindow.AddHandler(PointerPressedEvent, Button_OnPointerPressed, RoutingStrategies.Bubble);
    }

    private void Button_OnPointerPressed(object? sender, PointerPressedEventArgs e)
    {
        mainWindow.BeginMoveDrag(e);
    }
}

```
MainWindow.axaml

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:ComboBox_Test.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="ComboBox_Test.Views.MainWindow"
        x:DataType="vm:MainWindowViewModel"
        Icon="/Assets/avalonia-logo.ico"
		x:Name="mainWindow"
        Title="ComboBox_Test">

    <Design.DataContext>
        <vm:MainWindowViewModel/>
    </Design.DataContext>

    <Panel>
        <ComboBox Width="100" HorizontalAlignment="Center" VerticalAlignment="Center">
            <ComboBoxItem>Inline Item</ComboBoxItem>
        </ComboBox>
    </Panel>
</Window>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/135464720/26a6f6c6-df54-4443-8747-07ff27e7d6eb)

Follow the code:
https://github.com/henrrymoraes83/qtp/releases/download/Avalonia/ComboBox-Test-master.zip

**Comment by timunie** [comment]
This is by design. It's not the added handler that has the problem, but your invoke of `mainWindow.BeginMoveDrag(e);`. This will start to move the window around and mark the event as handled internally. So it will not be forwarded to the ComboBox. What's your reason to call this function at all in the ComboBox? 

https://github.com/AvaloniaUI/Avalonia/blob/f6809ad4e6b1f68d7cb1d797e5bd0d147910ce36/src/Avalonia.Controls/Window.cs#L412-L415

As soon as I comment the line in doubt, it works as expected: 
```cs
private void Button_OnPointerPressed(object? sender, PointerPressedEventArgs e)
{
    // mainWindow.BeginMoveDrag(e);
    Debug.WriteLine(e.ToString());
}
```


> Note: I think this is not an actual bug but more a question. Therefore I'll convert this into a discussion for now.

**Comment by henrrymoraes83** [comment]
Tim,

Thank for your reply.

I my case the Window has ExtendClientAreaChromeHints="NoChrome"and ExtendClientAreaToDecorationsHint="True", for this, moving the window is an important thing.

How can I move the window without using "AddHandler" + "BeginMoveDrag" and ensure that ComboBox works fine?

Please, do you know another way to do that?

**Reply by stevemonaco** [reply]
Normally, you create a small region for the titlebar and apply the event to that. If you want the entirety of the window (except controls) to be draggable, you can try to filter the events like the following. I'm not sure how well it scales to a real application though. You might want to add special cases for non-interactive components like `TextBlock` so it triggers a window drag, too.

```xml
<Grid x:Name="capture" Background="Blue">
    <ComboBox
        Width="100"
        HorizontalAlignment="Center"
        VerticalAlignment="Center">
        <ComboBoxItem>Inline Item</ComboBoxItem>
    </ComboBox>
</Grid>
```

```cs
public MainWindow()
{
    InitializeComponent();
    capture.AddHandler(PointerPressedEvent, Capture_OnPointerPressed, RoutingStrategies.Bubble);
}

private void Capture_OnPointerPressed(object? sender, PointerPressedEventArgs e)
{
    var control = this.InputHitTest(e.GetPosition(this));
    if (control == capture)
        this.BeginMoveDrag(e);
}
```

**Reply by henrrymoraes83** [reply, marked as answer]
Stephen. 

It worked fine. Thank you so much!!

**Comment by henrrymoraes83** [comment]
The moving Window not work. I put a breakpoint inside if (control == loginWindow) on this.BeginMoveDrag(e); and it has never enter here.

axaml
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:vm="clr-namespace:MyProject.ViewModels"
        xmlns:assets="clr-namespace:MyProject.Assets"
        mc:Ignorable="d" 
        x:Class="MyProject.LoginView"
        Title="LoginView"
        WindowState="Normal" WindowStartupLocation="CenterScreen"
        Background="Transparent"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaToDecorationsHint="True"
        x:Name="loginWindow"
        IsVisible="{Binding IsViewVisible}">


```
axaml.cs

```
public partial class LoginView : Window
{
    public LoginView()
    {
        InitializeComponent();
        loginWindow.AddHandler(PointerPressedEvent, Button_OnPointerPressed, RoutingStrategies.Bubble);
    }

    private void Button_OnPointerPressed(object? sender, PointerPressedEventArgs e)
    {
        Debug.WriteLine("Button_OnPointerPressed Pressed");
        var control = this.InputHitTest(e.GetPosition(this));
        if (control == loginWindow)
            this.BeginMoveDrag(e);
    }
```

**Reply by stevemonaco** [reply]
Right. Open DevTools and check the Visual Tree tab. You'll see that there's a `Panel` right below the Window. I think this is created internally by Avalonia and you can't avoid it. You could possibly do a type check for `Panel` in addition to the `loginWindow`, but that's moving forward into the hacky territory.

This is why I used a `Grid` and captured mouse events acting upon the `Grid`.
